test_unstructured_ingest/unit/pipeline/reformat/test_chunking.py

```diff
@@ -0,0 +1,144 @@
+        self, _ingest_docs_map_: Mock, caplog: LogCaptureFixture
+    ):
+        chunking_config = ChunkingConfig(chunking_strategy="by_invalid")
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig(partition_by_api=True)
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+        chunker.run(ELEMENTS_JSON_FILE)
+        assert "Input should be 'basic', 'by_page', 'by_similarity'" in caplog.text
+
+    def it_warns_with_nonlocal_chunking_strategy_and_partition_by_api_False(
+        self, _ingest_docs_map_: Mock, caplog: LogCaptureFixture
+    ):
+        chunking_config = ChunkingConfig(chunking_strategy="by_similarity")
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig(partition_by_api=True)
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+
+        chunker.run(ELEMENTS_JSON_FILE)
+        assert "There is no locally available chunking_strategy:" in caplog.text
+
+    # -- Chunker.chunk() ---------------------------------------------------------------------------
+
+    def it_skips_chunking_if_strategy_is_None(self):
+        chunking_config = ChunkingConfig(chunking_strategy=None)
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig()
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+        assert chunker.chunk(ELEMENTS_JSON_FILE) is None
+
+    # -- integration test --
+    @pytest.mark.parametrize("strategy", ["by_title", "basic"])
+    def it_chunks_locally(self, strategy: str, _ingest_docs_map_: Mock):
+        chunking_config = ChunkingConfig(chunking_strategy=strategy)
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig()
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+        chunked_elements = chunker.chunk(ELEMENTS_JSON_FILE)
+        assert all(isinstance(elem, CompositeElement) for elem in chunked_elements)  # type: ignore
+
+    def it_chunks_remotely(self, _ingest_docs_map_: Mock, _partition_via_api_: Mock):
+        chunking_config = ChunkingConfig(chunking_strategy="by_similarity")
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig(partition_by_api=True, api_key="aaaaaaaaaaaaaaaaaaaaa")
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+
+        chunker.chunk(ELEMENTS_JSON_FILE)
```

Comment:
All tests have three parts, setting up the test-fixture, running the unit under test, and then verifying the results. There are different names for it around like Arrange-Act-Assert and others you can find around.

I find it very helpful to the reader to separate the three sections with a blank line (and keep the sections together with no blank lines but sometimes comment lines.

The first question when reading a test or trying to find the test you're interested in is always: What is the unit under test? (unit being function or method generally). Making the line that exercises the unit stand out by being surrounded by a blank line above and below eases that search.